variable/shape.cpp

```diff
@@ -159,6 +159,94 @@ Variable reshape(const VariableConstView &view, const Dimensions &dims) {
   return reshaped;
 }
 
+void validate_fold_dims(const Dimensions &old_dims, const Dim from_dim,
+                        const Dimensions &to_dims) {
+  if (!old_dims.contains(from_dim))
+    throw except::DimensionError("Dimension to fold not found.");
+  // Make sure that new dims do not already exist in data dimensions,
+  // apart from the old dim (i.e. old dim can be re-used)
+  for (const auto dim : to_dims.labels())
+    if (old_dims.contains(dim) && dim != from_dim)
+      throw except::DimensionError(
+          "Fold: new dimensions cannot contain labels that already exist.");
```

Comment:
Why is this nessary? Isn't `fold_dims` doing this implicitly?

Reply:
Do you mean that `addInner` protects against duplicate dimensions? Yes, i forgot about that.

Reply:
Yes.